src/classes/AFFL_AccRecordType_TDTM.cls

```diff
@@ -156,13 +157,15 @@ public class AFFL_AccRecordType_TDTM extends TDTM_Runnable {
                 }
             }
             //A.2 - Change all affiliations that are not primary any more.
-            if(contactIDs.size() > 0 && accountIDs.size() > 0) {
+            if( contactIDs.size() > 0 && accountIDs.size() > 0) {
                 List<Affiliation__c> afflsMadePrimary = [select Contact__c, Account__c, Affiliation_Type__c
                           from Affiliation__c where Contact__c IN :contactIDs and Account__c IN :accountIDs];
                 AFFL_MultiRecordType_TDTM afflMulti = new AFFL_MultiRecordType_TDTM();
                 afflMulti.uncheckOtherPrimariesSameType(afflsMadePrimary, dmlWrapper);
             }
         }
+
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update, true);
```

Comment:
@skosgi1 @lifewithryan  This might fix the bug, but it's not clear to me why. I'm a little concerned that we're setting another classes's recursion flag here, as we're in the AFFL_AccRecordType_TDTM class, not the CON_PrimaryAffls_TDTM class.

Is the bug happening in the CON_PrimaryAffls_TDTM class as a result of that trigger logic being called when this logic here executes? Maybe we need logic in the CON_PrimaryAffls_TDTM class to detect duplicates somehow?

Reply:
@robertbwatson @lifewithryan  When the record type on the account changes the AFFL_AccRecordType_TDTM class updates the primary affiliation field on the Contact Object which would then fire CON_PrimaryAffls_TDTM to create another affiliation. So I am trying to stop firing CON_PrimaryAffls_TDTM trigger logic at the end of AFFL_AccRecordType_TDTM execution. Please let me know if the solution sounds ok.

Reply:
I think I might need to pull this down and do some testing of it tomorrow or Friday. The CON_PrimaryAffls_TDTM trigger has a lot of logic so I'd be concerned we're skipping it when we shouldn't necessarily be always skipping it.

I think there are other scenarios we might need to test here too, for example if we change an account's record type to a record type that isn't mapped to a Primary field on the Contact.

Reply:
Yea, let's deep dive into this one a bit. It seems like we shouldn't be turning off that trigger from within this one. If turning off that trigger works, then maybe there's logic in there that's to blame. If we have the re-entrancy code in the CON_PrimaryAffls_TDTM class already, then I would think that should kick in for us when it needs to.